src/main/java/com/github/netelli/TablesDAO.java

```diff
@@ -0,0 +1,39 @@
+package com.github.netelli;
+
+import org.apache.log4j.Logger;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public abstract class TablesDAO implements AutoCloseable {
```

Comment:
Methods like "createTable" and "insertData" are present in all ancestors of this class.
You need to declare them in this class and mark them as abstract (so they won't have an implementation, only declaration) -> otherwise it doesn't makes sense to have abstract class without abstract methods.
By declaring abstract methods you will force all ancestors to provide an implementation.

addtional:
mark all implementations of abstract methods with @Override annotation (for example method in ProductsDAO.createTable() should be marked so after applying review comments)